ffcx/element_interface.py

```diff
@@ -24,45 +20,9 @@ def convert_element(element: ufl.finiteelement.FiniteElementBase) -> basix.ufl._
     if isinstance(element, basix.ufl._ElementBase):
         return element
     else:
-        return _cached_conversion(element)
-
-
-@lru_cache()
-def _cached_conversion(element: ufl.finiteelement.FiniteElementBase) -> basix.ufl._ElementBase:
-    """Create an FFCx element from a UFL element.
-
-    Args:
-        element: A UFL finite element
-
-    Returns:
-        A Basix finite element
-    """
-    warnings.warn(
-        "Use of elements created by UFL is deprecated. You should create elements directly using Basix.",
-        DeprecationWarning)
-
-    # Tackle compositional elements, e.g. VectorElement first, then elements
-    # implemented by FFCx, then finally elements convertible by Basix.
-    if hasattr(ufl, "VectorElement") and isinstance(element, ufl.VectorElement):
-        return basix.ufl.blocked_element(
-            _cached_conversion(element.sub_elements()[0]), shape=(element.num_sub_elements(), ))
-    elif hasattr(ufl, "TensorElement") and isinstance(element, ufl.TensorElement):
-        if len(element.symmetry()) == 0:
-            return basix.ufl.blocked_element(_cached_conversion(element.sub_elements()[0]), shape=element._value_shape)
-        else:
-            assert element.symmetry()[(1, 0)] == (0, 1)
-            return basix.ufl.blocked_element(_cached_conversion(element.sub_elements()[0]),
-                                             element._value_shape, symmetry=True)
-    elif hasattr(ufl, "MixedElement") and isinstance(element, ufl.MixedElement):
-        return basix.ufl.mixed_element([_cached_conversion(e) for e in element.sub_elements()])
-    elif hasattr(ufl, "EnrichedElement") and isinstance(element, ufl.EnrichedElement):
-        return basix.ufl.enriched_element([_cached_conversion(e) for e in element._elements])
-    elif element.family() == "Quadrature":
-        return QuadratureElement(element.cell().cellname(), element.value_shape(), scheme=element.quadrature_scheme(),
-                                 degree=element.degree())
-    elif element.family() == "Real":
-        return RealElement(element)
-    else:
```

Comment:
So the reason this is not needed anymore is due to everything being in basix (where all caching now happens)?
If so, this is rather neat

Reply:
Yes. One of the main reasons for this PR was this this code is duplicated from code that already exists in Basix, except with the addition of QuadratureElement and RealElement.

Reply:
Except Basix doesn't cache. Creating elements in UFL (instead of directly with basix.ufl) is deprecated. Leaving in part of the function so that we could cache the result of a deprecated function seemed unnecessary as it'll be deleted soon anyway

Reply:
It does? https://github.com/FEniCS/basix/blob/7daa8bf1eb8774e685a6e30a91e585e409f03d87/python/basix/ufl.py#L1585

Reply:
Sort of. It caches pure Basix elements, but not Mixed, Quadrature, etc. Although the pure Basix elements will be the most costly to recreate so I guess this is good enough